CDP4Reporting/DataSource/ReportingDataSourceCategory.cs

```diff
@@ -63,22 +64,84 @@ protected ReportingDataSourceCategory()
 
+        /// <summary>
+        /// Checks if the <see cref="ElementDefinition.Category"/>, or <see cref="ElementUsage.Category"/> contains a <see cref="Category"/>
+        /// that has <see cref="ReportingDataSourceCategory{T}.ShortName"/> as its <see cref="Category.ShortName"/>, or somewhere in the
+        /// <see cref="Category.SuperCategory"/> hierarchy.
+        /// </summary>
+        /// <returns>
+        /// True if <see cref="Category.ShortName"/> was found, otherwise false.
+        /// </returns>
+        private bool HasCategory()
+        {
             var definitionCategory = this.Node.ElementDefinition.Category
                 .SingleOrDefault(x => x.ShortName == this.ShortName);
 
             if (definitionCategory != null)
             {
-                this.Value = true;
+                return true;
             }
 
-            var usageCategory = this.Node.ElementUsage?.Category
-                .SingleOrDefault(x => x.ShortName == this.ShortName);
+            foreach (var category in this.Node.ElementDefinition.Category)
+            {
+                if (this.ExistsInSuperCategory(category, null))
+                {
+                    return true;
+                }
+            }
 
-            if (usageCategory != null)
+            if (this.Node.ElementUsage != null)
             {
-                this.Value = true;
+                var usageCategory = this.Node.ElementUsage.Category
+                    .SingleOrDefault(x => x.ShortName == this.ShortName);
+
+                if (usageCategory != null)
+                {
+                    return true;
+                }
+
+                foreach (var category in this.Node.ElementUsage.Category)
+                {
+                    if (this.ExistsInSuperCategory(category, null))
+                    {
+                        return true;
+                    }
+                }
             }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Checks if a <see cref="Category"/> has has <see cref="ReportingDataSourceCategory{T}.ShortName"/> as its <see cref="Category.ShortName"/>
+        /// in its <see cref="Category.SuperCategory"/> hierarchy.
+        /// </summary>
+        /// <returns>
+        /// True if <see cref="Category.ShortName"/> was found, otherwise false.
+        /// </returns>
+        private bool ExistsInSuperCategory(Category category, List<Category> checkedCategories)
+        {
+            if (checkedCategories == null)
+            {
+                checkedCategories = new List<Category>();
+            }
+
+            foreach (var superCategory in category.SuperCategory.Except(checkedCategories))
+            {
+                if (superCategory.ShortName.Equals(this.ShortName))
+                {
+                    return true;
+                }
+
+                checkedCategories.Add(superCategory);
+
+                return this.ExistsInSuperCategory(superCategory, checkedCategories);
```

Comment:
The return here stop the foreach from iterating over all the super categories; it will effectively check only the first one.